### DISCUSSION TITLE
Bring the features upstream

### DISCUSSION THREAD

**Discussion by yoshimo** [Cookie-AutoDelete/Cookie-AutoDelete#1271] [Ideas / Feature Requests]
Looking at the drama of featured extensions  in https://github.com/Cookie-AutoDelete/Cookie-AutoDelete/issues/758
i was wondering wether CAD needs to remain a standalone third party extension.

Firefox itself can clean up various kinds of data on shutdown already.
Bringing in the easier "whitelist" / "greylist" options from CAD and the clean storage on closing a tab feature would make it almost equal by default.

How about upstreaming cookie auto delete code to firefox itself and stop worrying about api calls and featured extensions on mobile /desktop completely?

**Comment by kennethtran93** [comment]
The code for firefox itself is using a different programming language (C# or C++?) than WebExtension APIs (JavaScript).  If you know how to code for Firefox, feel free.

FYI - this webextension is also available on Google Chrome and Microsoft Edge Chromium.
